crdt/rlist.go

```diff
@@ -926,20 +950,92 @@ func (l *RList) DeleteCharAt(i int) error {
 	return l.DeleteChar()
 }
 
+// +-----------------------------------+
+// | Operations - Insert str container |
+// +-----------------------------------+
+
+//Inserts a string container as a child of the root atom.
+type InsertStr struct{}
+
+func (v InsertStr) AtomPriority() int { return 30 }
+func (v InsertStr) MarshalJSON() ([]byte, error) {
+	return json.Marshal("insert str container")
+}
+
+func (v InsertStr) String() string { return "STR: " }
+
+func (v InsertStr) ValidateChild(child AtomValue) error {
+	switch child.(type) {
+	case InsertChar:
+		return nil
+	case Delete:
+		return nil
+	default:
+		return fmt.Errorf("invalid atom value after InsertStr: %T (%v)", child, child)
+	}
+}
+
+// InsertStr inserts a Str container after the cursor position and advances the cursor.
+func (l *RList) InsertStr() error {
+	l.Cursor = AtomID{}
+	atomID, err := l.addAtom(InsertStr{})
+	l.Cursor = atomID
+	return err
+}
+
 // +------------+
 // | Conversion |
 // +------------+
 
-// AsString interprets list as a sequence of chars.
-func (l *RList) AsString() string {
+// ToString interprets list as a sequence of chars.
+func (l *RList) ToString() string {
 	atoms := l.filterDeleted()
 	chars := make([]rune, len(atoms))
 	for i, atom := range atoms {
-		chars[i] = atom.Value.(InsertChar).Char
+		if _, ok := atom.Value.(InsertStr); ok {
+			chars[i] = '*'
+		} else if _, ok := atom.Value.(InsertChar); ok {
+			chars[i] = atom.Value.(InsertChar).Char
+		}
 	}
 	return string(chars)
 }
 
+// this interface represents a generic type.
+type generic interface{}
+
+// ToJSON interprets list as a JSON.
+func (l *RList) ToJSON() ([]byte, error) {
+	tab := "    "
+	atoms := l.filterDeleted()
+	var elements []generic
+	for i := 0; i < len(atoms); {
+		currentAtomValue := atoms[i].Value
+		switch currentAtomValue.(type) {
```

Comment:
You can "capture" the casted value of a type switch with

```go
switch value := currentAtomValue.(type) { ... }
```

The type of `value` is different for each `case`. This will simplify other instances of this body, noted below